### PR TITLE
Move the Xdebug admin warning to Site Health

### LINKED ISSUE #3499: Improve Xdebug warning
## Feature description

As mentioned on #3198 and #1915, there are some issues with the current admin notice that is shown when Xdebug is active. Screenshot:

Screenshot 2019-10-14 at 16 07 09

Mainly, the notice is shown even when it is not actually relevant to the site (i.e. Reader mode is used) or when the user cannot do something about it (i.e. they are not an admin).

I'm seeing that myself as well as I've become numb to that warning and don't even notice it anymore really.

A  few thoughts:

1. Don't display notice when using Reader mode (at least until #2202 is done)
1. Only display notice when user can `manage_options`
1. Only display the notice on the AMP plugin settings screen
1. Make notice dismissible with persistence in user settings (bonus: still always show it on the settings screen)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Remove notice in favor or Site Health, or at least limit notice to just AMP settings screen.
* Limit to administrators and/or when `WP_DEBUG` is enabled?

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by kienstra**
## Summary
This removes the Xdebug admin notice:
removes-ad

...and moves it to Site Health:

### 1. Xdebug loaded, `WP_DEBUG` is undefined or falsey:
xdebug-loaded-wp-debug-false

### 2. Xdebug loaded, `WP_DEBUG` is `true`:

xdebug-loaded-wp-debug-true

### 3. Xdebug not loaded

xdebug-not-loaded

Fixes #3499

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by kienstra** [comment]
This is ready for review, but this isn't a high priority, as it's not milestoned for this release.

**Comment by westonruter** [comment]
After 6b87048a4daf7c4f3275fd839120688ab9bbee69:

# Xdebug enabled and WP_DEBUG disabled

image

# Xdebug enabled and WP_DEBUG enabled

image

# Xdebug disabled

image

**Comment by westonruter** [comment]
Build for testing: [amp.zip](https://github.com/ampproject/amp-wp/files/4376972/amp.zip) (v1.5.0-alpha-20200324T181257Z-c59dedecd)

File: includes/deprecated.php [hunk 1]
```diff
@@ -250,3 +250,27 @@ function amp_post_meta_box() {
...
+			<?php
+			esc_html_e(
+				'Your server currently has the Xdebug PHP extension loaded. This can cause some of the AMP plugin\'s processes to timeout depending on your system resources and configuration. Please deactivate Xdebug for the best experience.',
+				'amp'
+			);
+			?>
+		</p>
+	</div>
+	<?php
+}
```
Comment (kienstra) [inline includes/deprecated.php]: Ah, good catch moving it to `deprecated.php`. Sorry for missing that.

**Comment by westonruter** [comment]
For some reason Travis CI is not reporting the successful build, but I see ✅so I'm mergin'.

**Pull request merged**